src/map.cpp

```diff
@@ -6726,13 +6727,38 @@ void map::remove_rotten_items( Container &items, const tripoint &pnt )
     const tripoint abs_pnt = getabs( pnt );
     for( auto it = items.begin(); it != items.end(); ) {
         if( has_rotten_away( *it, abs_pnt ) ) {
+            //If the item that is rotting is a food item, see if we can spawn a monster
+            if (it->is_comestible()){
+                rotten_item_spawn(it, pnt);
+            }
+            //Remove the item regardless
             it = i_rem( pnt, it );
         } else {
             ++it;
         }
     }
 }
 
+template <typename Item>
+void map::rotten_item_spawn( Item &item, const tripoint &pnt )
```

Comment:
Why is this a function template? I guess it's because `item` is not really an item, but an iterator into a `std::list<item>` (which explains why it's dereferenced on every access even through it is a reference).

Change the function to take an actual item reference (and make it a reference to a `const item` as the function does not modify the item) and call it as `rotten_item_spawn( *it, ... )`.